Failed tests name contains `.` instead of `::` in verbose mode

When a test fails and pytest is run in verbose mode, the failed test name contains `.` instead of `::` which doesn't match the name expected by pytest itself to run the test.

e.g. the following code:

```python
import pytest


class TestFails:

    def test_fail1(self):
        assert True is False

    @pytest.mark.parametrize('foo,bar', [(1, 3), (2, 4)])
    def test_fail2(self, foo, bar):
        assert False is True
```

causes the following output:

```
$ pytest -v test_foo.py
Test session starts (platform: linux, Python 3.6.5, pytest 3.8.0, pytest-sugar 0.9.1)
cachedir: .pytest_cache
rootdir: /home/ack/Desktop, inifile:
plugins: xdist-1.23.0, sugar-0.9.1, postgresql-1.3.4, mock-1.10.0, forked-0.2, cov-2.6.0


―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― TestFails.test_fail1 ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

self = <test_foo.TestFails object at 0x7f9351761fd0>

    def test_fail1(self):
>       assert True is False
E       assert True is False

test_foo.py:7: AssertionError

 test_foo.py::TestFails.test_fail1 ⨯                                                                                                                                                                                                                                                                                                                                                                                        33% ███▍      

――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― TestFails.test_fail2[1-3] ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

self = <test_foo.TestFails object at 0x7f9356ee2748>, foo = 1, bar = 3

    @pytest.mark.parametrize('foo,bar', [(1, 3), (2, 4)])
    def test_fail2(self, foo, bar):
>       assert False is True
E       assert False is True

test_foo.py:11: AssertionError

 test_foo.py::TestFails.test_fail2[1-3] ⨯                                                                                                                                                                                                                                                                                                                                                                                   67% ██████▋   

――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― TestFails.test_fail2[2-4] ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

self = <test_foo.TestFails object at 0x7f9356eef2b0>, foo = 2, bar = 4

    @pytest.mark.parametrize('foo,bar', [(1, 3), (2, 4)])
    def test_fail2(self, foo, bar):
>       assert False is True
E       assert False is True

test_foo.py:11: AssertionError

 test_foo.py::TestFails.test_fail2[2-4] ⨯                                                                                                                                                                                                                                                                                                                                                                                  100% ██████████

Results (0.07s):
       3 failed
         - test_foo.py:6 TestFails.test_fail1
         - test_foo.py:9 TestFails.test_fail2[1-3]
         - test_foo.py:9 TestFails.test_fail2[2-4]

```

--- a/pytest_sugar.py
+++ b/pytest_sugar.py
@@ -361,13 +361,14 @@ class SugarTerminalReporter(TerminalReporter):
         else:
             fspath = report.fspath
         basename = os.path.basename(fspath)
-        if self.showlongtestinfo:
-            test_location = report.location[0]
-            test_name = report.location[2]
-        else:
-            test_location = fspath[0:-len(basename)]
-            test_name = fspath[-len(basename):]
         if print_filename:
+            if self.showlongtestinfo:
+                test_location = report.location[0]
+                test_name = report.location[2]
+            else:
+                test_location = fspath[0:-len(basename)]
+                test_name = fspath[-len(basename):]
+            test_name = test_name.replace('.', '::')
             self.current_lines[path] = (
                 " " +
                 colored(test_location, THEME['path']) +

--- a/test_sugar.py
+++ b/test_sugar.py
@@ -447,6 +447,25 @@ class TestTerminalReporter(object):
             output
         )
 
+    def test_verbose_has_double_colon_with_class(self, testdir):
+        testdir.makepyfile(
+            """
+            import pytest
+
+            class TestTrue:
+
+                def test_true():
+                    assert True
+            """
+        )
+        output = testdir.runpytest(
+            '--force-sugar', '--verbose'
+        ).stdout.str()
+
+        test_name = (
+            'test_verbose_has_double_colon_with_class.py::TestTrue::test_true')
+        assert test_name in strip_colors(output)
+
     def test_xdist(self, testdir):
         pytest.importorskip("xdist")
         testdir.makepyfile(

